### DISCUSSION TITLE
Visual Studio debug - keeps the dotnet process open after stopping

### DISCUSSION THREAD

**Discussion by markgr** [AvaloniaUI/Avalonia#12462] [General]
So this maybe a stupid question and may have been answered elsewhere but hopefully I am not the only one with it.

Ok so I have hidden the Normal window minimize, maximise close etc - my main window looks like this

![image](https://github.com/AvaloniaUI/Avalonia/assets/2209069/15b29dbe-9683-464e-957a-b1b642ecec22)

I Have a custom button to close in my 'title' area

```
private void Button_OnClick(object? sender, RoutedEventArgs e)
    {
        if (Avalonia.Application.Current.ApplicationLifetime is IClassicDesktopStyleApplicationLifetime desktop)
        {
            desktop.Shutdown();
        }
    }
```

And in my startup code I am setting up the shutdown mode as OnExplicitShutdown ( I have tried all three options).

Now whenever I start the application - it all runs just fine - but when I stop the app - either by clicking the Stop Debugging in VS - or by Clicking my button as above - the app does stop - however the Dotnet process continues and then I have to manually kill it otherwise I can't do a build as the Process has the application DLL in use....

Any Clues?

**Comment by timunie** [comment]
maybe anything is not yet GC-finished? Would be great if you can try to reproduce in a minimal sample and file it here if possible to reproduce. Also please let us know your OS.

**Reply by markgr** [reply]
Let me throw a simple app together and then I'll upload it to GitHub

It's Visual Studio 2022 on Windows 10

**Comment by thevortexcloud** [comment]
Do you use the previewer? It runs your full app in the background.

**Reply by markgr** [reply]
Interesting - let me take a look at that

**Reply by thevortexcloud** [reply]
I should note, it should not cause it to stop future builds. But I figured it's worth mentioning since it may be doing weird things.
